src/test/test_crypto.c

```diff
@@ -152,8 +152,13 @@ test_crypto_openssl_version(void *arg)
   const char *h_version = crypto_openssl_get_header_version_str();
   tt_assert(version);
   tt_assert(h_version);
-  tt_assert(!strcmpstart(version, h_version)); /* "-fips" suffix, etc */
-  tt_assert(!strstr(version, "OpenSSL"));
+  if (strcmpstart(version, h_version)) { /* "-fips" suffix, etc */
+    TT_DIE(("OpenSSL library version %s did not begin with header version %s.",
+            version, h_version));
+  }
+  if (strstr(version, "OpenSSL")) {
+    TT_DIE(("assertion failed: !strstr(\"%s\", \"OpenSSL\")", version));
+  }
   int a=-1,b=-1,c=-1;
   if (!strcmpstart(version, "LibreSSL") || !strcmpstart(version, "BoringSSL"))
     return;
```

Comment:
This has nothing to do with your patch since it was already a bug in the unittest, but if `!strstr(version, "OpenSSL")` passes above, then this would only be true here if someone named their "OpenSSL" library `LibreSSLOpenSSL-6.6.6` or `BoringSSL-OpenSSL-42` or something, which… would be very weird. I think we're pretty unlikely to hit this line.